src/front-end/typescript/lib/components/virtualized-table/index.tsx

```diff
@@ -0,0 +1,343 @@
+  | ADT<"scrollToTop", null>
+  | ADT<"scrollCompleted", null>;
+
+export const init: component_.base.Init<Params, State, Msg> = ({
+  idNamespace,
+  THView = Table.DefaultTHView,
+  TDView = Table.DefaultTDView,
+  totalItems,
+  rowHeight,
+  bufferSize = DEFAULT_BUFFER_SIZE
+}) => {
+  const initialEnd = Math.min(INITIAL_VISIBLE_ITEMS, totalItems);
+  return [
+    {
+      idNamespace,
+      THView,
+      TDView,
+      activeTooltipThIndex: null,
+      activeTooltipTdIndex: null,
+      visibleRange: { start: 0, end: initialEnd },
+      scrollTop: 0,
+      containerHeight: 0,
+      totalItems,
+      rowHeight,
+      bufferSize,
+      scrollContainerId: `virtualized-table-${idNamespace}-scroll-container`
+    },
+    []
+  ];
+};
+
+export const update: component_.base.Update<State, Msg> = ({ state, msg }) => {
+  switch (msg.tag) {
+    case "toggleTooltipTh": {
+      const currentThIndex = state.activeTooltipThIndex;
+      if (currentThIndex === null) {
+        return [state.set("activeTooltipThIndex", msg.value), []];
+      } else {
+        return [state.set("activeTooltipThIndex", null), []];
+      }
+    }
+    case "toggleTooltipTd": {
+      return [state, []];
+    }
+    case "handleScroll": {
+      const { scrollTop, containerHeight } = msg.value;
+      const rowHeight = state.rowHeight;
+      const buffer = state.bufferSize;
+      const start = Math.floor(scrollTop / rowHeight);
+      const visibleStart = Math.max(0, start - buffer);
+
+      if (containerHeight <= 0) {
+        return [state, []];
+      }
+
+      const visibleCount = Math.ceil(containerHeight / rowHeight);
+      const visibleEnd = Math.min(
+        state.totalItems || 0,
+        visibleStart + visibleCount + buffer * BUFFER_MULTIPLIER
+      );
+
+      return [
+        state
+          .set("scrollTop", scrollTop)
+          .set("containerHeight", containerHeight)
+          .set("visibleRange", { start: visibleStart, end: visibleEnd }),
+        []
+      ];
+    }
+    case "updateVisibleRange": {
+      return [
+        state.set("visibleRange", {
+          start: msg.value.start,
+          end: msg.value.end
+        }),
+        []
+      ];
+    }
+    case "updateTotalItems": {
+      const newTotalItems = msg.value;
+      const currentRange = state.visibleRange;
+      // Update visible range if current range is invalid with new total
+      const updatedRange = {
+        start: currentRange.start,
+        end: Math.min(currentRange.end, newTotalItems)
+      };
+      return [
+        state
+          .set("totalItems", newTotalItems)
+          .set("visibleRange", updatedRange),
+        []
+      ];
+    }
+    case "resetScroll": {
+      const initialEnd = Math.min(INITIAL_VISIBLE_ITEMS, state.totalItems || 0);
+      return [
+        state
+          .set("scrollTop", 0)
+          .set("visibleRange", { start: 0, end: initialEnd }),
+        [component_.cmd.delayedDispatch(10, adt("scrollToTop", null))]
```

Comment:
This seems to be doing nearly the same thing as init; will the component still work if we add this command to init an call init wherever we're dispatching nested resetScroll messages? i.e., when handling onInitResponse and search messages in src/front-end/typescript/lib/pages/user/list.tsx